### ISSUE TITLE
Sort similarity scores

### ISSUE THREAD

**Issue by nbgl** [data61/anonlink-entity-service#335]
Similarity scores need to be sorted before solving. There are some helper tools in `anonlink.concurrency` for this.

**Comment by nbgl** [comment]
I’ve disabled two tests in `test_result_correctness.py` to account for the output of anonlink v0.11 being correct, but the Entity Service being still incorrect. They need to be reenabled.

**Comment by hardbyte** [comment]
This was incorrectly close by a comment in https://github.com/data61/anonlink-entity-service/pull/339

**Comment by nbgl** [comment]
The greedy solver needs to consider scores from highest to lowest to maximise accuracy. Sorting before solving is the usual way of doing this (I describe a more efficient way in data61/anonlink#212, but it is more complex to implement).

The Entity Service _does_ actually sort similarity scores now, so this issue may be closed. The code is [here](https://github.com/data61/anonlink-entity-service/commit/ffb2c7a8700b3c3945190321886f5145367369ed?diff=unified#diff-30fb0765378c9296147bce2d345daed9R239) as part of #339.

A concern might be that this aggregation is single-threaded, so it might not be the most efficient. Parallelising this merge sort is not a research question, but a software engineering one. (Not hard, but annoying.) But this should be identified as a bottleneck before any work is done, and it should be a separate issue.

**Comment by hardbyte** [comment]
Thanks for the clarification Jakub
